### DISCUSSION TITLE
Update to latest

### DISCUSSION THREAD

**Discussion by tjboldt** [tjboldt/Apple2-IO-RPi#65] [Announcements]
## Update to latest

Note that these steps assume your system is fairly recent. If you have issues, you might want to start over setting up from scratch, starting at step 4 in #63 

1. Restart Apple II
2. `-SHELL`
3. `cd /home/pi/Apple2-IO-RPi`
4. `git pull`
5. `cd RaspberryPi/apple2driver`
6. `go build`
7. `sudo reboot`
8. Restart Apple II
9. `-UPDATE.FIRMWARE` (assuming you have default drive as Drive 1)
10. Enter slot number the card is in
11. Wait for firmware update to complete all four pages

**Comment by Abysmal** [comment]
Step #2 - SHould this not be Run the 'SHELL" command -SHELL ?

**Reply by tjboldt** [reply]
Thanks for pointing that out. The select run command was for the older firmware that had a menu and built in pseudo shell.

**Comment by SalvadorCastello** [comment]
Please, do you know how long does it take the step "go build" in the script "setup.sh"?
I have a Pi Zero 2W and it's running for more than 2 hours... it's burning the chip.
Thanks

**Reply by tjboldt** [reply]
Normally it takes a minute or so although I have had it take many minutes longer the first build. It won’t break anything to hit ^C and try again if it seems stuck.

**Reply by SalvadorCastello** [reply]
I have tried 4 times , each one I needed to boot the PI because it had frozen , not accepting any command.
One question: To run the setup the PI MUST be plugged to the board (and to the Apple2) as said ?
I'm running the script out of the Apple2.
Thanks

**Reply by SalvadorCastello** [reply]
After some "echo" added to the setup.sh script I found that the problem was on git clone apple2-io-rpi , I solved with this command before running the script: git config --global pack.threads "1"

I have 3 internet links, I don't know if this can be interfering .
Anyway the scrip went till the end.

Thanks

**Reply by tjboldt** [reply]
I haven't had to change that setting for any of my builds so you might be onto something with it being an environment issue. I'm glad you found a fix.

**Reply by SalvadorCastello** [reply]
It's working now on my Apple IIe Platinum.
Now I'll begin to try everything.
Thanks for your replies.
